tools/logicalimager/RegVal.cpp

```diff
@@ -157,6 +163,7 @@ int RegVal::initialize(const Rejistry::RegistryValue *value) {
             // This shouldn't happen because we check the range above.
             break;
         }
+        delete valueData;
```

Comment:
Again,  I don't think value->getValue() made a copy, so you shouldn't delete valueData

Reply:
Yes it does. VLD stacktrace
> 
---------- Block 4086905 at 0x000000003594E920: 24 bytes ----------
  Leak Hash: 0x41B99ABE, Count: 10, Total 240 bytes
  Call Stack:
    ntdll.dll!RtlAllocateHeap()
    f:\dd\vctools\crt\vcstartup\src\heap\new_scalar.cpp (19): tsk_logical_imager.exe!operator new() + 0xA bytes
    c:\users\jkho\git\sleuthkit\rejistry++\src\registryvalue.cpp (34): tsk_logical_imager.exe!Rejistry::RegistryValue::RegistryValue() + 0xA bytes
    c:\users\jkho\git\sleuthkit\tools\logicalimager\regval.cpp (129): tsk_logical_imager.exe!RegVal::initialize() + 0x42 bytes
    c:\users\jkho\git\sleuthkit\tools\logicalimager\regparser.cpp (240): tsk_logical_imager.exe!RegParser::getValue() + 0x12 bytes
    c:\users\jkho\git\sleuthkit\tools\logicalimager\registryanalyzer.cpp (252): tsk_logical_imager.exe!RegistryAnalyzer::analyzeSAMUsers() + 0x2E bytes
    c:\users\jkho\git\sleuthkit\tools\logicalimager\tsk_logical_imager.cpp (579): tsk_logical_imager.exe!main()
    f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl (65): tsk_logical_imager.exe!invoke_main()
    f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl (253): tsk_logical_imager.exe!__scrt_common_main_seh() + 0x5 bytes
    f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl (296): tsk_logical_imager.exe!__scrt_common_main()
    f:\dd\vctools\crt\vcstartup\src\startup\exe_main.cpp (17): tsk_logical_imager.exe!mainCRTStartup()
    KERNEL32.DLL!BaseThreadInitThunk() + 0x14 bytes
    ntdll.dll!RtlUserThreadStart() + 0x21 bytes
  Data:
    08 E0 AF C3    F6 7F 00 00    60 14 8D 35    B8 01 00 00     ........ `..5....
    A4 19 00 00    CD CD CD CD   
>